### PR TITLE
Upgrade to Java 21

### PR DESCRIPTION
**Pull request by ebyhr**
Fixes https://github.com/trinodb/trino-gateway/issues/79
https://github.com/trinodb/trino/pull/20010 upgraded Java to 21 in Trino

### CODE REVIEW

**Comment by mosabua** [comment]
Nice .. do we want to upgrade now, wait until Trino is on Java 21 or wait even longer? Wdyt @vishalya @willmostly @Chaho12 

Maybe we can change it to be able to build with Java 21 (upgrade dependency) .. but not require it yet.

**Review by mosabua — DISMISSED** [review]
Imho we should at least wait until Trino requires Java 21 to require Java 21 for Trino Gateway.

File: docs/release-notes.md [hunk 1]
```diff
@@ -50,6 +50,6 @@ Gateway](https://github.com/lyft/presto-gateway/) 1.9.5 codebase
 * Add PostgreSQL support for backend database. (https://github.com/trinodb/trino-gateway/pull/13)
 * Allow routing of `/v1/node` endpoint URL. (https://github.com/trinodb/trino-gateway/pull/27)
 * Filter logs for sensitive information. (https://github.com/trinodb/trino-gateway/pull/9)
-* Require Java 17 for build and runtime. (https://github.com/trinodb/trino-gateway/pull/16)
+* Require Java 21 for build and runtime. (https://github.com/trinodb/trino-gateway/pull/16)
```
Comment (mosabua) [inline docs/release-notes.md:53]: This is wrong

**Comment by Chaho12** [comment]
I agree, Trino doesn't officially require Java 21 as of now, and not many users are using java 21 so I think we should wait for a while.
Are there any big performance improvement with new version?

**Comment by mosabua** [comment]
I think we should pull this PR apart and merge the update of the dependency to allow building with Java 21 .. and then update to Java 21 after a release or two

**Comment by mosabua** [comment]
Cherri picked mvel upgrade into https://github.com/trinodb/trino-gateway/pull/156

**Comment by mosabua** [comment]
We should look at updating and verifying dependencies first .. or at least rebase and test this. Team discussion is positive about going to Java 21 in line with Trino soon though.

**Pull request closed without merging**